client/landing/gutenboarding/components/domain-picker-button/index.tsx

```diff
@@ -3,41 +3,83 @@
  */
 import * as React from 'react';
 import classnames from 'classnames';
+import { useSelect } from '@wordpress/data';
 import { Icon, chevronDown } from '@wordpress/icons';
+import { sprintf } from '@wordpress/i18n';
+import { useI18n } from '@automattic/react-i18n';
 
 /**
  * Internal dependencies
  */
-import Link, { Props as LinkProps } from '../link';
+import Link from '../link';
 import { usePath, Step } from '../../path';
+import { STORE_KEY as ONBOARD_STORE } from '../../stores/onboard';
+import { DOMAIN_SUGGESTIONS_STORE } from '../../stores/domain-suggestions';
 
 /**
  * Style dependencies
  */
 import './style.scss';
 
-interface Props extends LinkProps {
-	children: string | React.ReactElement;
-}
-
-const DomainPickerButton: React.FunctionComponent< Props > = ( {
-	children,
-	className,
-	...buttonProps
-} ) => {
+const DomainPickerButton: React.FunctionComponent = () => {
+	const { __, i18nLocale } = useI18n();
 	const makePath = usePath();
+	const { domain, domainSearch, siteTitle, siteVertical } = useSelect( ( select ) =>
+		select( ONBOARD_STORE ).getState()
+	);
+
+	// Use site title or vertical as search query for a domain suggestion
+	const suggestionQuery = siteTitle || siteVertical?.label;
+	const domainSuggestion = useSelect(
+		( select ) => {
+			// Get suggestion only if the query is valid and if there isn't a selected domain
+			if ( ! suggestionQuery || suggestionQuery.length < 2 || domain ) {
```

Comment:
Testing that `suggestionQuery` still has something if everything else but alpha-numerics are removed would be nice addition, but nothing urgent. You can try emojis or some Japanese for example. :-)

Reply:
You mean [this issue](https://github.com/Automattic/wp-calypso/issues/40748)? 😄  
I'd be happy to take it in the next sprint as a follow up of current update.